### PR TITLE
fix: use tldts for subdomain-agnostic domain matching [LIVE-25164]

### PR DESCRIPTION
**Pull request by Justkant**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.**
- [x] **Impact of the changes:**
  - Buy/sell and earn domain check logic

### 📝 Description

- Use tldts library's getDomain function
- Implement subdomain matching in isSameDomain (app.example.com now matches api.example.com)
- Optimize URL parsing with shared parseUrl helper to create single URL object
- Remove unused getDomain and getProtocol helper functions
- Add test coverage for:
  - Complex TLDs (.co.uk, etc.)
  - Localhost and IP address edge cases
  - Multiple subdomain levels
  - Base domain to subdomain matching

https://github.com/user-attachments/assets/167200ee-f339-499b-b739-fbc896144278

### ❓ Context

- **JIRA or GitHub link**: [LIVE-25164]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-25164]: https://ledgerhq.atlassian.net/browse/LIVE-25164?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts [hunk 1]
```diff
@@ -137,6 +106,14 @@ describe("manifestDomainUtils", () => {
     it("should return false when both URLLs are invalid", () => {
       expect(isSameDomain("not-a-url", "not-a-url")).toBe(false);
     });
+
+    it("should return false for localhost URLs", () => {
+      expect(isSameDomain("http://localhost:3000", "http://localhost:8080")).toBe(false);
```
[the remark this answers is not in the document]
Comment (Justkant) [reply libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts:111]: localhost is not a valid TLD, so tldts returns null
